### DISCUSSION TITLE
[help] How to get current pipeline overall seconds used?

### DISCUSSION THREAD

**Discussion by CorradoLanera** [ropensci/targets#1289] [Help] [answered]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

This is to ask what the suggested way is to calculate the overall amount of time the current (successful) pipeline spends on execution.

My current idea is the following:

```r
tar_meta(fields = c(seconds, warnings, error)) |>
  dplyr::filter(is.na(warnings), is.na(error), !is.na(seconds)) |>
  dplyr::pull(seconds) |>
  sum()
```

But I'd like to know if it will count targets that are no longer in the pipeline but still stored on meta (i.e., the ones that would be removed by `tar_prune()`).
Or there could be a more straightforward way to obtain that with a command I did not know/remember. The main point is to keep the "original" time skipped targets spent to be evaluated, too.

Thank you!
Corrado.

## UPDATE after answer
So the suggested solution is:

```r
tar_meta(
    tar_manifest()[["name"]],
    fields = c(seconds, warnings, error)
  ) |>
  dplyr::filter(is.na(warnings), is.na(error), !is.na(seconds)) |>
  dplyr::pull(seconds) |>
  sum()
```

Thank you!

**Comment by wlandau-lilly** [comment, marked as answer]
This is a bit of a subtly tricky questions because of how [parallel computing](https://books.ropensci.org/targets/crew.html) dynamically schedules the targets as parallel workers become available, but what you propose should be fine if you restrict the `tar_meta()` output to the targets listed by `tar_manifest()$name`.
